suites/mux/muxer_suite.go

```diff
@@ -401,23 +424,18 @@ func SubtestStreamReset(t *testing.T, tr mux.Multiplexer) {
 		}
 
 		s.Close()
-		done <- struct{}{}
 	}()
 
 	muxb, err := tr.NewConn(b, false)
 	if err != nil {
 		t.Fatal(err)
 	}
 
-	go func() {
-		str, err := muxb.AcceptStream()
-		checkErr(t, err)
-		str.Reset()
-		done <- struct{}{}
-	}()
+	str, err := muxb.AcceptStream()
+	checkErr(t, err)
+	str.Reset()
 
-	<-done
-	<-done
+	wg.Wait()
```

Comment:
Otherwise, we close the connections early. However, if we _don't_ reach this point, we _want_ to close the connections before we wait.